core/src/main/scala/kafka/server/KafkaApis.scala

```diff
@@ -3927,6 +3929,36 @@ class KafkaApis(val requestChannel: RequestChannel,
     }
   }
 
+  def handleShareFetchRequest(request: RequestChannel.Request): Unit = {
+    val shareFetchRequest = request.body[ShareFetchRequest]
+
+    if (!config.isNewGroupCoordinatorEnabled) {
```

Comment:
Could we rely on whether the protocol is enabled instead of using `isNewGroupCoordinatorEnabled`? We're moving away from it. See https://github.com/apache/kafka/commit/ba61ff0cd94dd50401998fff1be7f646d1239000.

Reply:
I was wondering how group.version would affect this. Thanks, @dajac.

Reply:
I think the check for new consumer group would be much sensible for ShareGroupDescribe and ShareGroupHeartbeat and rather less for fetch and acknowledge. I have removed the code for now from here.

Reply:
I think that you’ll need a single source for both.   If I remember correctly, we said in the KIP that we will use group.version.